synchronization/timer.go

```diff
@@ -0,0 +1,31 @@
+package synchronization
```

Comment:
I'd document the motivation for this type, including a reference to the golang issue with the standard timer, so that if that issue goes away we can kill our own impl

Reply:
sure, good idea